source/server/hot_restarting_child.cc

```diff
@@ -7,6 +7,42 @@ namespace Server {
 
 using HotRestartMessage = envoy::HotRestartMessage;
 
+void HotRestartingChild::UdpForwardingContext::registerListener(
+    const Network::Address::Instance& address, Network::UdpListenerConfig& listener_config) {
+  bool inserted =
+      listener_map_.try_emplace(address.asString(), ForwardEntry{&address, &listener_config})
+          .second;
+  ASSERT(inserted, "Two udp listeners on the same address shouldn't be possible");
+}
+
+absl::optional<HotRestartingChild::UdpForwardingContext::ForwardEntry>
+HotRestartingChild::UdpForwardingContext::getListenerForDestination(
+    const Network::Address::Instance& address) {
+  auto it = listener_map_.find(address.asString());
+  if (it == listener_map_.end()) {
+    // If no listener on the specific address was found, check for a default route.
+    // If the address is IPv6, check default route IPv6 first, otherwise check default
+    // route IPv4, but either way check both because default route should still apply
+    // even if only one listener was declared.
```

Comment:
If only 0.0.0.0 is configured, I don't think any IPv6 packets can be received. On the other hand, if the socket is configured with v4 compact, configuring a `::` listener should also listen on v4 address.

Reply:
Thanks, I found all sorts of unclear and contradictory information about the behavior of default routes with different protocols. Changed to match this (though I don't think I have scope to detect v4 compact).

Reply:
There is an interface `Ipv6::v6only()`.

Reply:
Aha, thanks! Incorporated.